DatadogSessionReplay/Tests/Recorder/RecordingCoordinatorTests.swift

```diff
@@ -164,56 +180,13 @@ class RecordingCoordinatorTests: XCTestCase {
 
         // When
         rumContextObserver.notify(rumContext: .mockRandom())
+        recordingTriggerMock.triggerCallback?()
 
         // Then
         let metric = try XCTUnwrap(telemetry.messages.last?.asMetric)
         XCTAssertEqual(metric.name, "Method Called")
     }
 
-    // MARK: StartRecordingImmediately Initialization Tests
-
-    func test_whenStartRecordingImmediatelyIsDefault_itShouldRecord() throws {
```

Comment:
That's interesting point. It's mostly about lack of ability to test `SessionReplayFeature` properly, because after the refactor - that's the object responsible for calling it.

IMO we should revisit how we serve all the dependencies to it, and make it testable. Ideally in a separate PR.

Reply:
Entire thing happens in `SessionReplayFeature` and that's the object responsible for starting the recording immediately when `configuration.startRecordingImmediately` is set to `true`. We can move this responsibility down to `RecordingCoordinator`, but it will only check if the new implementation is working.

In order to test `SessionReplayFeature` we would need to introduce dependency factory to be able to inject a mock. This way we could check that if `startRecordingImmediately == true`,  `SessionReplayFeature` calls `startRecording()` on `recordingCoordinator`.

Reply:
I was able to re-enable this coverage in 9019d8d4a47416356005d7a53603ef0c87508149 by simulating start/stop with triggers 👍  I also fixed a flaky tests

Reply:
@maciejburda it looks like you forgot to re-add these tests?

Reply:
Yes, these are missing again 🤔